Source/LinqToDB/Common/StringBuilderExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace LinqToDB.Common
+{
+	public static class StringBuilderExtensions
+	{
+		private static readonly uint[] _lookup32 = CreateLookup32();
+        
+		private static uint[] CreateLookup32()
+		{
+			var result = new uint[256];
+			for (int i = 0; i < 256; i++)
+			{
+				string s =i.ToString("X2");
+				result[i] = ((uint)s[0]) + ((uint)s[1] << 16);
+			}
+			return result;
+		}
+        
+		/// <summary>
+		/// Appends an array of bytes to a <see cref="StringBuilder"/> in hex (i.e. 255->FF)
+		/// format utilizing a static lookup table to minimize allocations.
+		/// </summary>
+		/// <param name="sb">The <see cref="StringBuilder"/> to append to</param>
+		/// <param name="bytes">The byte array to append in hex</param>
+		/// <remarks>
+		/// The implementation here was chosen based on:
+		/// https://stackoverflow.com/a/624379/2937845
+		/// Which indicated that  https://stackoverflow.com/a/24343727/2937845's
+		/// implementation of ByteArrayToHexViaLookup32 was the fastest method
+		/// not involving unsafe 
+		/// </remarks>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		public static void AppendByteArrayAsHexViaLookup32(this StringBuilder sb, byte[] bytes)
+		{
+			var lookup32 = _lookup32;
+			for (int i = 0; i < bytes.Length; i++)
+			{
+				var val = lookup32[bytes[i]];
+				sb.Append((char)val);
+				sb.Append((char) (val >> 16));
+			}
```

Comment:
Can you please review and compare if this might be better? I believe that there are optimizations the JIT does on a `foreach` on an array vs using an index.
```suggestion
			foreach (var b in bytes
			{
				var val = lookup32[b];
				sb.Append((char)val);
				sb.Append((char) (val >> 16));
			}
```

Reply:
It looks like it's either a wash or a little bit worse. 

On one hand, the generated ASM is essentially identical or only one byte better for the `foreach` version.

On the other hand, `foreach` comes in with ~6 more bytes of IL. I tossed both versions into sharplab:

https://sharplab.io/#gist:360c05b61e74e8c6ecdec7a3ee34797c

Reply:
Regarding benchmarks: You should always use Benchmark.NET to run benchmarks. The library is written to correctly test micro-benchmarks and has tools to manage various things that can affect benchmark performance. Here's an updated copy of your code w/ Benchmark.NET: http://share.linqpad.net/x3j7fk.linq On my machine, the `foreach` is .1us faster, but within the margin of error. I would say it is worth the difference (given how negligible the difference is) in either direction to have more idiomatic code (i.e. use the `foreach`), but that's my personal opinion.

Reply:
>I would say it is worth the difference (given how negligible the difference is) in either direction to have more idiomatic code (i.e. use the foreach), but that's my personal opinion.

I'm fine with switching it to `foreach` given the lack of difference, but if we are adding something like @MaceWindu 's suggested `MaxBinaryParameterLogSize` for SqlText logging purposes we may need to stick with the for loop

Reply:
> 
> 
> Regarding benchmarks: You should always use Benchmark.NET to run benchmarks. The library is written to correctly test micro-benchmarks and has tools to manage various things that can affect benchmark performance. Here's an updated copy of your code w/ Benchmark.NET: http://share.linqpad.net/x3j7fk.linq 

BTW thank you for showing me how to use Benchmark.NET in Linqpad... IDK why I never tried it that way.